# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/SystemPrincipal.cpp
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 caps/SystemPrincipal.cpp
--- a/caps/SystemPrincipal.cpp	Tue May 21 22:51:54 2019 +0000
+++ b/caps/SystemPrincipal.cpp	Tue May 21 23:14:27 2019 +0000
@@ -52,40 +52,6 @@
 }
 
 NS_IMETHODIMP
-SystemPrincipal::GetCsp(nsIContentSecurityPolicy** aCsp) {
-  *aCsp = nullptr;
-  return NS_OK;
-}
-
-NS_IMETHODIMP
-SystemPrincipal::SetCsp(nsIContentSecurityPolicy* aCsp) {
-  // Never destroy an existing CSP on the principal.
-  // This method should only be called in rare cases.
-
-  return NS_ERROR_FAILURE;
-}
-
-NS_IMETHODIMP
-SystemPrincipal::EnsureCSP(dom::Document* aDocument,
-                           nsIContentSecurityPolicy** aCSP) {
-  // CSP on a system principal makes no sense
-  return NS_ERROR_FAILURE;
-}
-
-NS_IMETHODIMP
-SystemPrincipal::GetPreloadCsp(nsIContentSecurityPolicy** aPreloadCSP) {
-  *aPreloadCSP = nullptr;
-  return NS_OK;
-}
-
-NS_IMETHODIMP
-SystemPrincipal::EnsurePreloadCSP(dom::Document* aDocument,
-                                  nsIContentSecurityPolicy** aPreloadCSP) {
-  // CSP on a system principal makes no sense
-  return NS_ERROR_FAILURE;
-}
-
-NS_IMETHODIMP
 SystemPrincipal::GetDomain(nsIURI** aDomain) {
   *aDomain = nullptr;
   return NS_OK;